spec.html

```diff
@@ -8034,11 +8034,10 @@ <h1>ArraySetLength ( _A_, _Desc_ )</h1>
             1. Set _newLenDesc_.[[Writable]] to *true*.
           1. Let _succeeded_ be ! OrdinaryDefineOwnProperty(_A_, `"length"`, _newLenDesc_).
           1. If _succeeded_ is *false*, return *false*.
-          1. Repeat, while _newLen_ &lt; _oldLen_,
-            1. Set _oldLen_ to _oldLen_ - 1.
-            1. Let _deleteSucceeded_ be ! _A_.[[Delete]](! ToString(_oldLen_)).
+          1. For each own property key _P_ of _A_ that is an array index and is greater than or equal to _newLen_, in descending numeric index order, do
```

Comment:
A property key (a String in this case) cannot be greater than or equal to a Number, so it'd have to be something like

> For each own property key _P_ of _A_ that is an array index, whose numeric value is greater than or equal to _newLen_